packages/contracts-bedrock/contracts/periphery/OpStackChainRegistry.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+/**
+ * @title  OpStackChainRegistry
+ * @notice OpStackChainRegistry acts as a way to register and query
+ *         contract addresses for an OP Stack chain
+ */
+contract OpStackChainRegistry {
+
+    error OnlyDeploymentAdmin();
+
+    /**
+     * @notice Struct representing a deployment
+     *         on an OP Stack chain
+     */
+    struct Deployment {
```

Comment:
shower thought: Maybe people should also have to claim a `chainId` as well as the name of the deployment? rfc @tynes @smartcontracts

Reply:
Hmmm is chain ID part of system config?

Reply:
It is not as far as I can tell.

Reply:
I feel like if we cannot answer this question, this contract is far from being used in production